acados/ocp_nlp/ocp_nlp_common.c

```diff
@@ -2930,6 +2930,71 @@ void ocp_nlp_res_get_inf_norm(ocp_nlp_res *res, double *out)
 }
 
 
+// NOTE: if NaNs are in residual vector, blasfeo_inf_norm might return 0.
```

Comment:
The NaN case was not considered when implementing the infinity norm routine.
Namely, the routine is currently implemented using fmax, which doesn't propagate NaN.
It can be reimplemented using proper checks such that NaN is propagated.

More specifically, one way to check for NaN (also in your code below) is to check whether
`if(res_entry!=res_entry) then res_entry is NaN;`
This works on IEEE floating point standard complying compilers.
Otherwise, in C99 it is defined the function `isnan()`.

Reply:
yes, I can also just do the `!=` comparison instead if you prefer that.
I thought, just checking for NaN when exiting SQP because of the residuals is ok.
But I can also directly check for it in res_compute if you have no concerns regarding performance.
Or do you want to change `VECNRM_INF` in Blasfeo?

Reply:
I can update the `vecnrm_inf` in BLASFEO, if this PR can wait a few more days.

I didn't get the concern about performance: how would checking `!=` instead affect performance?

Reply:
Ok, I will wait until you update the BLASFEO master.

> I didn't get the concern about performance: how would checking != instead affect performance?

I just meant checking in every residual computation vs only when exiting SQP.
But I agree checking `!=` is less confusing.